webapp/test/test_updates.py

```diff
@@ -169,3 +169,72 @@ def test_process_list_empty_json(self):
             self.updates_api.process_list(1, UPDATES_JSON_EMPTY.copy())
         assert "'package_list' is a required property" in str(context)
+
+class TestSplayTree(TestBase):
+    """ Set of tests for Splay tree implementation. """
+    hot_cache = HotCache()
+    test_nevra = []
+
+    @pytest.fixture(autouse=True)
+    def setup_dummy_data(self):
+        """ Create list with dummy nevra data for the tree. """
+        self.test_nevra.append({"key": 1, "content": {"response": "nevra1"}})
+        self.test_nevra.append({"key": 2, "content": {"response": "nevra2"}})
+        self.test_nevra.append({"key": 3, "content": {"response": "nevra3"}})
+        self.test_nevra.append({"key": 4, "content": {"response": "nevra4"}})
+        self.test_nevra.append({"key": 5, "content": {"response": "nevra5"}})
+        self.test_nevra.append({"key": 6, "content": {"response": "nevra6"}})
+
+    @pytest.fixture
+    def setup_small_tree(self):
+        """ Fixture creates small tree for testing. """
+        self.hot_cache.insert(self.test_nevra[2]["key"], self.test_nevra[2]["content"])
+        self.hot_cache.insert(self.test_nevra[0]["key"], self.test_nevra[0]["content"])
+        self.hot_cache.insert(self.test_nevra[1]["key"], self.test_nevra[1]["content"])
+
+    @pytest.fixture
+    def setup_huge_tree(self):
+        """ Fixture creates huge tree for pruning tests, etc. """
+        for order in range(0, 1025):
+            self.test_nevra[0]["key"] = order
+            self.hot_cache.insert(self.test_nevra[0]["key"], self.test_nevra[0]["content"])
+
+
+    def test_insertion_only_one(self):
+        """ Test only one insertion and if its in the tree. """
+        self.hot_cache.insert(self.test_nevra[0]["key"], self.test_nevra[0]["content"])
+
+        test_node = self.hot_cache.find(self.test_nevra[0]["key"])
+        assert test_node == self.test_nevra[0]["content"]
+
+    def test_find_nonexistent(self):
+        """ Test for finding nonexistent key. """
+        assert self.hot_cache.find(10) is None
+
+    def test_insertion_more(self, setup_small_tree):
+        """ Test for insertion more keys. """
+        test_node = self.hot_cache.find(self.test_nevra[0]["key"])
+        assert test_node == self.test_nevra[0]["content"]
+
+        test_node = self.hot_cache.find(self.test_nevra[2]["key"])
+        assert test_node == self.test_nevra[2]["content"]
+
+        test_node = self.hot_cache.find(self.test_nevra[1]["key"])
+        assert test_node == self.test_nevra[1]["content"]
+
+    def test_insertion_count(self, setup_small_tree):
+        """ Test for counting inserts. """
+        assert self.hot_cache.inserts == 3
+
+    def test_update_key(self, setup_small_tree):
+        """ Test for updating node content in tree. """
+        updated_content = {"reponse": "updatednevra2"}
+
+        self.hot_cache.insert(self.test_nevra[2]["key"], updated_content)
+
+        test_node = self.hot_cache.find(self.test_nevra[2]["key"])
```

Comment:
As I see the tests, maybe it would be better to split your testing collection into `test_nevra_keys` and `test_nevra_contents`. It avoids permanent writing of `["keys"]` and `["content"]`.

Reply:
Ok, I see you don' like this my proposal :cry: .

Reply:
No, its not like that, but I just think that having the content and key together is more self explanatory for next code reader :). Also saving the few characters in line is not that ground breaking. (just my opinion)